modules/Bio/EnsEMBL/Utils/IO/FeatureSerializer.pm

```diff
@@ -0,0 +1,106 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+
+=head1 NAME
+
+FeatureSerializer - An abstract serializer for turning EnsEMBL Features into other formats
+
+=head1 SYNOPSIS
+
+my $serializer = new Bio::EnsEMBL::Utils::IO::FeatureSerializer( $filehandle );
+$serializer->print_feature_list( \@list_of_features );
+
+=head1 DESCRIPTION
+
+Generic class for serializing features Subclass this class to create a 
+format-specific serializer. Be sure to implement print_feature at the 
+bare minimum
+
+=cut
+
+package Bio::EnsEMBL::Utils::IO::FeatureSerializer;
+
+use strict;
+use warnings;
+
+use base qw(Bio::EnsEMBL::Utils::IO::Serializer);
+
+# Majority of methods inherited from Serializer
+
+=head2 print_feature
+
+	Arg []     : None
+	Description: Abstract method to print a feature.
+                     Implemented in derived classes.
+        Returntype : None
+
+=cut
+
+sub print_feature {
+	throw( "print_feature method not implemented.");
```

Comment:
Better take out these tabs while we're here. Sorry, the code is from before our whitespace policy.